cmd/minishift/cmd/root.go

```diff
@@ -333,3 +360,29 @@ func setDefaultActiveProfile() {
 		}
 	}
 }
+
+// validateProfile check if a profile exist or not when used --profile flag
+// If profile exist then it set constants.ProfileName as profile provided using --profile option
+func validateProfile(cmd *cobra.Command) {
+	if cmd.Parent() != nil {
+		// This condition true for each command execpt *m profile* and *m start ...*
+		if cmd.Parent().Name() != profileFlag && cmd.Name() != startFlag {
+			var profileName string
+			// Iterate over provided arguments check if --profile used.
+			// This can't be handled using viper.GetString(profileFlag) because minishift always
+			// going to be default profile name.
+			for i, arg := range os.Args {
+				if arg == "--"+profileFlag {
+					profileName = os.Args[i+1]
+					break
+				}
+			}
+			if profileName != "" {
+				if !cmdUtil.IsValidProfile(profileName) {
+					atexit.ExitWithMessage(1, fmt.Sprintf("Profile: %s doesn't exist, Use *minishift profile set %s* to create", profileName, profileName))
+				}
+				constants.ProfileName = profileName
```

Comment:
The function `validateProfile` is not reflecting this step. If I don't look into the code of validateProfile then, I will not know where my profile is getting set.
I would highly recommend to move this line with other similar line `constants.MachineName = constants.ProfileName`.